### PR TITLE
feat: adding RegisteredPayees to ics29 genesis state

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

- Adding `RegisteredPayees` to ICS29 `GenesisState`

Needs: #1491 
This PR includes the additions from the above and should be merged after the initial work in #1491 

**NOTE**: this is one of many PRs to complete the feature work referenced in: #1477 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: modules/apps/29-fee/types/genesis.go [hunk 1]
```diff
@@ -77,5 +85,20 @@ func (gs GenesisState) Validate() error {
 		}
 	}
 
+	// Validate DistributionAddresses
+	for _, registeredDistAddr := range gs.RegisteredDistributionAddresses {
+		if _, err := sdk.AccAddressFromBech32(registeredDistAddr.Address); err != nil {
```
Comment (AdityaSripal) [inline modules/apps/29-fee/types/genesis.go:90]: also for DistributionAddress

**Review by AdityaSripal — APPROVED** [review]

File: modules/apps/29-fee/keeper/keeper_test.go [hunk 2]
```diff
@@ -257,3 +276,28 @@ func (suite *KeeperTestSuite) TestGetAllRelayerAddresses() {
 	suite.Require().Len(addr, len(expectedAddr))
 	suite.Require().Equal(addr, expectedAddr)
 }
+
+func (suite *KeeperTestSuite) TestGetAllDistributionAddresses() {
```
Comment (AdityaSripal) [inline modules/apps/29-fee/keeper/keeper_test.go:280]:
```suggestion
func (suite *KeeperTestSuite) TestGetAllPayeeAddresses() {
```

File: proto/ibc/applications/fee/v1/genesis.proto [hunk 3]
```diff
@@ -16,11 +16,14 @@ message GenesisState {
   // list of fee enabled channels
   repeated FeeEnabledChannel fee_enabled_channels = 2
       [(gogoproto.moretags) = "yaml:\"fee_enabled_channels\"", (gogoproto.nullable) = false];
+  // list of registered payees
+  repeated RegisteredPayee registered_payees = 3
+      [(gogoproto.moretags) = "yaml:\"registered_payees\"", (gogoproto.nullable) = false];
   // list of registered relayer addresses
-  repeated RegisteredRelayerAddress registered_relayers = 3
+  repeated RegisteredRelayerAddress registered_relayers = 4
```
Comment (AdityaSripal) [inline proto/ibc/applications/fee/v1/genesis.proto:23]: Why are there two of these? Shouldn't there be `RegisteredPayees` and `RegisteredCounterpartyPayees`?

**Review by chatton — APPROVED** [review]
LGTM nice work :1st_place_medal:

**Pull request merged**